src/map/map.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, Logger, HttpStatus } from '@nestjs/common';
+import { PrismaService } from '../prisma.service';
+import { ReadMapDto } from './dto/read-map.dto';
+import { HttpService } from '@nestjs/axios';
+import CustomException from 'src/exceptions/custom.exception';
+import { firstValueFrom } from 'rxjs';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class MapService {
+  constructor(
+    private prisma: PrismaService,
+    private readonly http: HttpService,
+    private readonly config: ConfigService,
+  ) {}
+
+  private readonly logger = new Logger(MapService.name);
+
+  async getMapById(
+    mapId: number,
+    lat: number,
+    lang: number,
+  ): Promise<ReadMapDto> {
+    try {
+      const map = await this.prisma.map.findUnique({
+        where: {
+          id: +mapId,
+        },
+      });
+
+      if (!map) {
+        throw new CustomException(HttpStatus.NOT_FOUND, 'Not Found - Map');
+      }
+
+      const defaultImage = this.config.get('DEFAULT_IMAGE');
+
+      const destinationLatitude = map.latitude;
+      const destinationLongitude = map.longitude;
+
+      const osrmRouteUrl = `http://router.project-osrm.org/route/v1/driving/${lang},${lat};${destinationLongitude},${destinationLatitude}`;
+      const osrmResponse = await firstValueFrom(this.http.get(osrmRouteUrl));
+      const distance = `${osrmResponse.data.routes[0].distance}m`;
```

Comment:
반환값 중 distance에 단위인 'm'를 더해 클라에게 넘겨줍니다

Reply:
osrmResponse.data 가 없는 경우는 아예 없나요? 
그런게 아니라면 `optional property` 사용이 좋아보입니다.

```suggestion
      const distance = `${osrmResponse.data?.routes[0]?.distance}m`;
```

이렇게 되면 거리가 나오지 않는 상황에서 m 를 어떻게 할지 고민해봐야겠군요.

Reply:
흠 데이터가 안넘어오는 경우도 있을 것 같아요. 확인해보겠습니다. 음 옵셔널로 넣으면 데이터가 안넘어올 때 'm'로 클라한테 넘겨질 것 같은데 디폴트 값이 있는게 좋을까요?

Reply:
근데 사실 거리가 없을리는 (?) 없어서 ... 없으면 직선거리라도 줘야하나 ?